### PR TITLE
arch: Move -fstack-protector-all and -finstrument-functions from Make.defs to Toolchain.defs 

### PR DESCRIPTION
**Pull request by xiaoxiang781216**
## Summary
Refactor makefile

## Impact
stack check

## Testing
Pass CI

### CODE REVIEW

File: arch/arm/src/armv7-m/Toolchain.defs [hunk 1]
```diff
@@ -77,6 +77,13 @@ else
   ARCHOPTIMIZATION += -fomit-frame-pointer
 endif
 
+ifeq ($(CONFIG_STACK_CANARIES),y)
+  ARCHOPTIMIZATION += -fstack-protector-all
```
Comment (pkarashchenko) [inline arch/arm/src/armv7-m/Toolchain.defs:81]:
```suggestion
  ARCHOPTIMIZATION += -fstack-protector-all
endif
```

File: arch/arm/src/armv8-m/Toolchain.defs [hunk 2]
```diff
@@ -77,6 +77,13 @@ else
   ARCHOPTIMIZATION += -fomit-frame-pointer
 endif
 
+ifeq ($(CONFIG_STACK_CANARIES),y)
+  ARCHOPTIMIZATION += -fstack-protector-all
```
Comment (pkarashchenko) [inline arch/arm/src/armv8-m/Toolchain.defs:81]:
```suggestion
  ARCHOPTIMIZATION += -fstack-protector-all
endif
```

Comment (xiaoxiang781216) [reply arch/arm/src/armv8-m/Toolchain.defs:81]: Done.

File: arch/arm/src/armv7-m/Toolchain.defs (hunk 1 above)
Comment (xiaoxiang781216) [reply arch/arm/src/armv7-m/Toolchain.defs:81]: Done.

**Review by pkarashchenko — APPROVED** [review]

**Pull request merged**